### PR TITLE
Use source language when specified

### PR DESCRIPTION
**Pull request by Muennighoff**
See [ch65394](https://app.clubhouse.io/dataiku/story/65394/maintenance-use-source-languages-when-specified)

### CODE REVIEW

**Review by arnaudde — COMMENTED** [review]
I could reproduce that source language was not taken into account. However I see no difference in the translation when using a random source language. The api does not send any detected source language but it seems to send the same translation. Did you observe any change ?

**Comment by Muennighoff** [comment]
Only rare cases, but e.g.:

```
text = "coin"
response = client.translate(text, target_language="ja", source_language=None)
>>> {'translatedText': 'コイン', 'detectedSourceLanguage': 'en', 'input': 'coin'}

text = "coin"
response = client.translate(text, target_language="ja", source_language="fr")
>>> {'translatedText': 'コーナー', 'input': 'coin'}
```

Also on a 100 row dataset, auto-detect en > af takes 2.2988691329956055s & non-auto en > af takes 1.9221627712249756s for me in DSS.

**Review by arnaudde — APPROVED** [review]
Good spot ! Fix looks good to me 👌 
I did not double check the small differences you mention in https://github.com/dataiku/dss-plugin-nlp-google-cloud-translation/pull/2#issuecomment-860609453 but I trust you on those.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## [Version 1.0.1](https://github.com/dataiku/dss-plugin-nlp-google-cloud-translation/releases/tag/v1.0.1) - Source language bug fix - 2021-06
```
Comment (alexcombessie) [inline CHANGELOG.md]:
```suggestion
## [Version 1.0.1](https://github.com/dataiku/dss-plugin-nlp-google-cloud-translation/releases/tag/v1.0.1) - Bugfix release - 2021-06
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+
+- Use source language when specified instead of always auto-detecting due to being faster and possible detection mistakes
```
Comment (alexcombessie) [inline CHANGELOG.md]:
```suggestion
- Fixed incorrectly ignored source language UI parameter (previously, it was always auto-detecting)
```

**Review by alexcombessie — CHANGES_REQUESTED** [review]
Can you bump the version in plugin.json?

**Comment by Muennighoff** [comment]
Versions should be good now

**Review by alexcombessie — APPROVED** [review]

**Pull request merged**